runtime/ast/unmarshal.go

```diff
@@ -0,0 +1,901 @@
+	var statementKindName string
+	if err := json.Unmarshal(b, &struct{ Type *string }{&statementKindName}); err != nil {
+		return err
+	}
+	if sk, ok := statementKindMap[statementKindName]; ok {
+		*a.Statement = sk()
+		return json.Unmarshal(b, &a.Statement)
+	}
+	return fmt.Errorf("unknown Statement Kind")
+}
+
+func (a *StatementsUnmarshaler) UnmarshalJSON(b []byte) error {
+	var rms []json.RawMessage
+	err := json.Unmarshal(b, &rms)
+	if err != nil {
+		return err
+	}
+	*a.Statements = make([]Statement, len(rms))
+	for i, v := range rms {
+		err := json.Unmarshal(v, &StatementUnmarshaler{&(*a.Statements)[i]})
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (a *TypeUnmarshaler) UnmarshalJSON(b []byte) error {
+	var typeName string
+	err := json.Unmarshal(b, &struct{ Type *string }{&typeName})
+	if err != nil {
+		return err
+	}
+	if f, ok := typeNameMap[typeName]; ok {
+		*a.Type = f()
+		return json.Unmarshal(b, &a.Type)
+	}
+	return fmt.Errorf("unknown Type")
+}
+
+//////////////////////////////////////////////////////////////////////////////
+
+func (a *Access) UnmarshalJSON(b []byte) error {
+	if c, ok := accessKindMap[strings.Trim(string(b), `"`)]; ok {
+		*a = c
+		return nil
+	}
+	return fmt.Errorf("unknown Access kind")
+}
+
+func (a *Argument) UnmarshalJSON(b []byte) error {
+	type Alias Argument
+	err := json.Unmarshal(b, &struct {
+		*Alias
+		Expression ExpressionUnmarshaler
+	}{
+		Alias:      (*Alias)(a),
+		Expression: ExpressionUnmarshaler{&a.Expression},
+	})
+	return err
+}
+
+func (a *AssignmentStatement) UnmarshalJSON(b []byte) error {
+	type Alias AssignmentStatement
+	return json.Unmarshal(b, &struct {
+		*Alias
+		Target   ExpressionUnmarshaler
+		Value    ExpressionUnmarshaler
+		StartPos *Position
+		EndPos   *Position
+	}{
+		Alias:  (*Alias)(a),
+		Target: ExpressionUnmarshaler{&a.Target},
+		Value:  ExpressionUnmarshaler{&a.Value},
+	})
+}
+
+func (a *Block) UnmarshalJSON(bs []byte) error {
+	type Alias Block
+	err := json.Unmarshal(bs, &struct {
+		*Alias
+		Statements StatementsUnmarshaler
+	}{
+		Alias:      (*Alias)(a),
+		Statements: StatementsUnmarshaler{&a.Statements},
+	})
+	return err
+}
+
+func (a *BreakStatement) UnmarshalJSON(b []byte) error {
+	type Alias BreakStatement
+	return json.Unmarshal(b, &struct {
+		*Alias
+	}{
+		Alias: (*Alias)(a)})
+}
+
+func (a *CompositeDeclaration) UnmarshalJSON(b []byte) error {
+	type Alias CompositeDeclaration
+	return json.Unmarshal(b, &struct {
+		*Alias
+	}{
+		Alias: (*Alias)(a)})
+}
+
+func (a *Condition) UnmarshalJSON(b []byte) error {
+	type Alias Condition
+	err := json.Unmarshal(b, &struct {
+		*Alias
+		Test    ExpressionUnmarshaler
+		Message ExpressionUnmarshaler
+	}{
+		Alias:   (*Alias)(a),
+		Test:    ExpressionUnmarshaler{&a.Test},
+		Message: ExpressionUnmarshaler{&a.Message},
+	})
+	return err
+}
+
+func (a *ConditionKind) UnmarshalJSON(b []byte) error {
+	if c, ok := conditionKindMap[strings.Trim(string(b), `"`)]; ok {
```

Comment:
Instead of trimming the serialized string's quotes manually, potentially leaving quoted content, unmarshal as a string:

```go
var s string
if err := json.Unmarshal(b, &s); err != nil {
	return err
}
```